client/landing/gutenboarding/analytics.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useOnUnmount } from 'landing/gutenboarding/hooks/use-on-unmount';
+import { StepNameType } from './path';
+
+export function recordLeaveStep( stepName: StepNameType ) {
+	recordTracksEvent( 'calypso_signup_step_leave', {
+		flow: 'gutenboarding',
```

Comment:
I don't know if it will help, but I created a [generic method](https://github.com/Automattic/wp-calypso/pull/41129/files#diff-616ab0addcbbbdb9a8afb3d28db448e6R18) over at #41129 that pre-fills the flow.

```
export function trackEventWithFlow( eventId: string, params = {}, flow = FLOW_ID ): void {
	recordTracksEvent( eventId, {
		flow,
		...params,
	} );
}
```

Reply:
ahh nice! that looks good.

Reply:
Feel free to change/erase anything on https://github.com/Automattic/wp-calypso/pull/41129 by the way. It's fairly hobbled together, and will probably have to sync with your work anyway 👍